Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2692,7 +2686,9 @@ namespace OpenInfraPlatform
 					// Interpret parameter
 					double sineTerm = sine->SineTerm * this->UnitConvert()->getLengthInMeterFactor();
 					//if length of the curve is less than 1, factor is equal to 1, otherways factor is equal to lenght 
-					double factor = length;
+					double factor;
+					if (length < 1)factor = 1;
+					else factor = length;

```

Comment:
Possibility to learn: in C syntax, this would b a one liner:

```suggestion
					double factor = length < 1.0 ? 1.0 : length;
```

Another possibility would be to use std::max:

```
					double factor = std::max(length, 1.0);
```

In any case, when assigning values to double variables, or comparing doubles, always use decimal notation - 1.0 instead of 1.

Reply:
Thank you for you suggestion. I chose the option two with `std::max()`. 
Resolved in [d580137](https://github.com/tumcms/Open-Infra-Platform/pull/553/commits/d580137f72d12e4dea27ab4c1f0936a7c180c8b0)